### PR TITLE
resource/instance: fix non-empty plan after refresh 

### LINKED ISSUE #13118: aws_instance dynamic ebs_block_device forces replacement
I ran into an issue with changes being forced on every run while attempting to deploy servers using a third-party AMI while enabling storage encryption. I used a dynamic block to mimic the `block_device_mappings` from the AMI, which produced a working encrypted system but Terraform detects the system as changed on every run even though the listed plan shows the same values for the removed and added blocks. Unlike the condition noted in the AWS instance docs, these instances do not mix  usage with `aws_ebs_volume`.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.12.24
+ provider.aws v2.59.0
+ provider.http v1.1.1
```
### Affected Resource(s)

* aws_instance

### Terraform Configuration Files

```hcl
  dynamic "ebs_block_device" {
    for_each = data.aws_ami.cis_amazon_linux2.block_device_mappings
    iterator = device
    content {
      device_name = device.value["device_name"]
      encrypted   = true
      kms_key_id  = aws_kms_key.app_server.arn
      iops        = device.value["ebs"]["iops"]
      snapshot_id = device.value["ebs"]["snapshot_id"]
      volume_size = device.value["ebs"]["volume_size"]
      volume_type = device.value["ebs"]["volume_type"]
    }
  }
```

### Expected Behavior

No changes were attempted

### Actual Behavior

```
      - ebs_block_device { # forces replacement
          - delete_on_termination = true -> null
          - device_name           = "/dev/sdb" -> null
          - encrypted             = true -> null
          - iops                  = 100 -> null
          - kms_key_id            = "arn:aws:kms:us-east-1:…:key/4c134819-e3f1-48e1-a9f3-078c51e50247" -> null
          - snapshot_id           = "snap-0141e981aeb73231b" -> null
          - volume_id             = "vol-03883d920c410e30a" -> null
          - volume_size           = 2 -> null
          - volume_type           = "gp2" -> null
        }
      + ebs_block_device { # forces replacement
          + delete_on_termination = true
          + device_name           = "/dev/sdb"
          + encrypted             = true
          + iops                  = (known after apply)
          + kms_key_id            = "arn:aws:kms:us-east-1:…:key/4c134819-e3f1-48e1-a9f3-078c51e50247"
          + snapshot_id           = "snap-0141e981aeb73231b"
          + volume_id             = (known after apply)
          + volume_size           = 2
          + volume_type           = "gp2"
        }
```

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13118

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/instance: set ebs_block_device from root_block_device when only 1 block device mapping in instance
```

Output from acceptance testing:

```
--- PASS: TestAccAWSInstanceDataSource_gp2IopsDevice (61.17s)
--- PASS: TestAccAWSInstanceDataSource_PlacementGroup (68.79s)
--- PASS: TestAccAWSInstanceDataSource_rootInstanceStore (82.66s)
--- PASS: TestAccAWSInstanceDataSource_VPCSecurityGroups (83.97s)
--- PASS: TestFetchRootDevice (0.00s)
    --- PASS: TestFetchRootDevice/device_name_in_mappings (0.00s)
    --- PASS: TestFetchRootDevice/device_name_not_in_mappings (0.00s)
    --- PASS: TestFetchRootDevice/no_images (0.00s)
--- PASS: TestAccAWSInstanceDataSource_EbsBlockDevice_KmsKeyId (91.01s)
--- PASS: TestAccAWSInstanceDataSource_privateIP (94.50s)
--- SKIP: TestAccAWSInstance_inEc2Classic (1.17s)
--- PASS: TestAccAWSInstanceDataSource_RootBlockDevice_KmsKeyId (105.64s)
--- PASS: TestAccAWSInstanceDataSource_creditSpecification (119.14s)
--- PASS: TestAccAWSInstanceDataSource_SecurityGroups (119.52s)
--- PASS: TestAccAWSInstanceDataSource_VPC (119.71s)
--- PASS: TestAccAWSInstanceDataSource_basic (129.03s)
--- PASS: TestAccAWSInstanceDataSource_GetUserData_NoUserData (130.24s)
--- PASS: TestAccAWSInstanceDataSource_blockDevices (131.20s)
--- PASS: TestAccAWSInstanceDataSource_getPasswordData_falseToTrue (144.76s)
--- PASS: TestAccAWSInstance_inDefaultVpcBySgName (64.02s)
--- PASS: TestAccAWSInstanceDataSource_getPasswordData_trueToFalse (156.45s)
--- PASS: TestAccAWSInstanceDataSource_GetUserData (164.53s)
--- PASS: TestAccAWSInstanceDataSource_keyPair (181.79s)
--- SKIP: TestAccAWSInstance_outpost (0.00s)
--- PASS: TestAccAWSInstance_GP2IopsDevice (70.20s)
--- PASS: TestAccAWSInstance_EbsBlockDevice_KmsKeyArn (90.81s)
--- PASS: TestAccAWSInstance_ipv6AddressCountAndSingleAddressCausesError (11.30s)
--- PASS: TestAccAWSInstance_rootInstanceStore (69.94s)
--- PASS: TestAccAWSInstance_inDefaultVpcBySgId (114.39s)
--- PASS: TestAccAWSInstance_GP2WithIopsValue (94.99s)
--- PASS: TestAccAWSInstanceDataSource_AzUserData (227.13s)
--- PASS: TestAccAWSInstance_basic (138.84s)
--- PASS: TestAccAWSInstanceDataSource_tags (242.31s)
--- PASS: TestAccAWSInstance_vpc (87.19s)
--- PASS: TestAccAWSInstance_disableApiTermination (98.83s)
--- PASS: TestAccAWSInstancesDataSource_tags (193.99s)
--- PASS: TestAccAWSInstance_ipv6_supportAddressCount (79.40s)
--- PASS: TestAccAWSInstancesDataSource_basic (205.21s)
--- PASS: TestAccAWSInstance_sourceDestCheck (127.11s)
--- PASS: TestAccAWSInstancesDataSource_instance_state_names (193.39s)
--- PASS: TestAccAWSInstance_NetworkInstanceSecurityGroups (78.93s)
--- PASS: TestAccAWSInstance_ipv6_supportAddressCountWithIpv4 (87.32s)
--- PASS: TestAccAWSInstance_NetworkInstanceRemovingAllSecurityGroups (84.89s)
--- PASS: TestAccAWSInstance_NetworkInstanceVPCSecurityGroupIDs (78.91s)
--- PASS: TestAccAWSInstanceDataSource_metadataOptions (317.58s)
--- PASS: TestAccAWSInstance_noAMIEphemeralDevices (175.81s)
--- PASS: TestAccAWSInstance_privateIP (67.28s)
--- PASS: TestAccAWSInstance_keyPairCheck (64.18s)
--- PASS: TestAccAWSInstance_multipleRegions (130.75s)
--- PASS: TestAccAWSInstance_rootBlockDeviceMismatch (64.98s)
--- PASS: TestAccAWSInstance_volumeTagsComputed (103.51s)
--- PASS: TestAccAWSInstance_volumeTags (113.70s)
--- PASS: TestAccAWSInstance_userDataBase64 (233.35s)
--- PASS: TestAccAWSInstance_tags (126.92s)
--- PASS: TestAccAWSInstance_EbsRootDevice_basic (74.19s)
--- PASS: TestAccAWSInstance_placementGroup (181.52s)
--- PASS: TestAccAWSInstance_withIamInstanceProfile (115.88s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifySize (113.07s)
--- PASS: TestAccAWSInstance_forceNewAndTagsDrift (134.05s)
--- PASS: TestAccAWSInstance_RootBlockDevice_KmsKeyArn (291.02s)
--- PASS: TestAccAWSInstance_primaryNetworkInterface (68.55s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyType (113.65s)
--- PASS: TestAccAWSInstance_primaryNetworkInterfaceSourceDestCheck (88.19s)
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifySize (107.13s)
--- PASS: TestAccAWSInstance_associatePublic_explicitPublic (67.50s)
--- PASS: TestAccAWSInstance_associatePublic_defaultPrivate (81.92s)
--- PASS: TestAccAWSInstance_associatePublicIPAndPrivateIP (186.98s)
--- PASS: TestAccAWSInstance_associatePublic_defaultPublic (88.08s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyIOPS (134.84s)
--- PASS: TestAccAWSInstance_associatePublic_explicitPrivate (76.13s)
--- PASS: TestAccAWSInstance_addSecurityGroupNetworkInterface (108.25s)
--- PASS: TestAccAWSInstance_addSecondaryInterface (117.03s)
--- PASS: TestAccAWSInstance_associatePublic_overridePublic (67.36s)
--- PASS: TestAccAWSInstance_associatePublic_overridePrivate (66.77s)
--- PASS: TestAccAWSInstance_CreditSpecification_UnspecifiedToEmpty_NonBurstable (68.14s)
--- PASS: TestAccAWSInstance_changeInstanceType (225.72s)
--- PASS: TestAccAWSInstance_creditSpecification_unknownCpuCredits_t2 (73.43s)
--- PASS: TestAccAWSInstance_creditSpecification_standardCpuCredits (91.18s)
--- PASS: TestAccAWSInstance_creditSpecification_unspecifiedDefaultsToStandard (104.50s)
--- PASS: TestInstanceTenancySchema (0.00s)
--- PASS: TestInstanceHostIDSchema (0.00s)
--- PASS: TestInstanceCpuCoreCountSchema (0.00s)
--- PASS: TestInstanceCpuThreadsPerCoreSchema (0.00s)
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices (193.30s)
--- PASS: TestAccAWSInstance_getPasswordData_falseToTrue (145.50s)
--- PASS: TestAccAWSInstance_creditSpecificationT3_standardCpuCredits (106.55s)
--- PASS: TestAccAWSInstance_creditSpecificationT3_unlimitedCpuCredits (102.13s)
--- PASS: TestAccAWSInstance_creditSpecification_updateCpuCredits (110.36s)
--- PASS: TestAccAWSInstance_getPasswordData_trueToFalse (150.46s)
--- PASS: TestAccAWSInstance_creditSpecification_unlimitedCpuCredits (140.46s)
--- PASS: TestAccAWSInstance_UserData_EmptyStringToUnspecified (92.93s)
--- PASS: TestAccAWSInstance_creditSpecification_isNotAppliedToNonBurstable (132.31s)
--- PASS: TestAccAWSInstance_creditSpecificationT3_updateCpuCredits (128.91s)
--- PASS: TestAccAWSInstance_UserData_UnspecifiedToEmptyString (96.42s)
--- PASS: TestAccAWSInstance_disappears (244.65s)
--- PASS: TestAccAWSInstance_hibernation (214.54s)
--- PASS: TestAccAWSInstance_CreditSpecification_Empty_NonBurstable (310.76s)
--- PASS: TestAccAWSInstance_creditSpecification_unknownCpuCredits_t3 (308.87s)
--- PASS: TestAccAWSInstance_creditSpecificationT3_unspecifiedDefaultsToUnlimited (310.23s)
--- PASS: TestAccAWSInstance_creditSpecification_unlimitedCpuCredits_t2Tot3Taint (390.69s)
--- PASS: TestAccAWSInstance_creditSpecification_standardCpuCredits_t2Tot3Taint (400.28s)
```
 Failing in TC:
```
--- FAIL: TestAccAWSInstance_blockDevices (54.66s)
--- FAIL: TestAccAWSInstance_metadataOptions (281.46s)
--- FAIL: TestAccAWSInstance_EbsRootDevice_ModifyAll (122.22s)
--- FAIL: TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination (92.74s)
--- FAIL: TestAccAWSInstance_instanceProfileChange (130.77s)
--- FAIL: TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination (100.62s)
```
But passing locally:
```
--- PASS: TestAccAWSInstance_blockDevices (103.57s)
--- PASS: TestAccAWSInstance_metadataOptions (128.92s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyAll (151.02s)
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination (122.33s)
--- PASS: TestAccAWSInstance_instanceProfileChange (221.20s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination (129.02s)

```

### CODE REVIEW

File: aws/resource_aws_instance.go [hunk 1]
```diff
@@ -655,21 +655,21 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
...
 
 	// Wait for the instance to become running so we can get some attributes
 	// that aren't available until later.
 	log.Printf(
 		"[DEBUG] Waiting for instance (%s) to become running",
-		*instance.InstanceId)
+		aws.StringValue(instance.InstanceId))
 
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{"pending"},
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:670]: For the instance state name values you could use the Go SDK constants

```go
const (
	// InstanceStateNamePending is a InstanceStateName enum value
	InstanceStateNamePending = "pending"

	// InstanceStateNameRunning is a InstanceStateName enum value
	InstanceStateNameRunning = "running"

	// InstanceStateNameShuttingDown is a InstanceStateName enum value
	InstanceStateNameShuttingDown = "shutting-down"

	// InstanceStateNameTerminated is a InstanceStateName enum value
	InstanceStateNameTerminated = "terminated"

	// InstanceStateNameStopping is a InstanceStateName enum value
	InstanceStateNameStopping = "stopping"

	// InstanceStateNameStopped is a InstanceStateName enum value
	InstanceStateNameStopped = "stopped"
)
```

defined in `github.com/aws/aws-sdk-go/service/ec2/api.go`.

Comment (anGie44) [reply aws/resource_aws_instance.go:670]: nice, thanks @ewbankkit! I'll update those

File: aws/resource_aws_instance.go [hunk 2]
```diff
@@ -843,13 +843,13 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.Set("ebs_optimized", instance.EbsOptimized)
-	if instance.SubnetId != nil && *instance.SubnetId != "" {
+	if instance.SubnetId != nil && aws.StringValue(instance.SubnetId) != "" {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:846]: `aws.StringValue` converts a `nil` input to `""`, so `if aws.StringValue(instance.SubnetId) != ""` should be sufficient here.

File: aws/resource_aws_instance.go [hunk 3]
```diff
@@ -1778,14 +1799,14 @@ func readBlockDeviceMappingsFromConfig(d *schema.ResourceData, conn *ec2.EC2) ([
 				ebs.VolumeType = aws.String(v)
 			}
 
-			if v, ok := bd["iops"].(int); ok && v > 0 && *ebs.VolumeType == "io1" {
+			if v, ok := bd["iops"].(int); ok && v > 0 && aws.StringValue(ebs.VolumeType) == "io1" {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:1802]: Can use `ec2.VolumeTypeIo1` here.

File: aws/resource_aws_instance.go [hunk 4]
```diff
@@ -1778,14 +1799,14 @@ func readBlockDeviceMappingsFromConfig(d *schema.ResourceData, conn *ec2.EC2) ([
...
 				// Only set the iops attribute if the volume type is io1. Setting otherwise
 				// can trigger a refresh/plan loop based on the computed value that is given
 				// from AWS, and prevent us from specifying 0 as a valid iops.
 				//   See https://github.com/hashicorp/terraform/pull/4146
 				//   See https://github.com/hashicorp/terraform/issues/7765
 				ebs.Iops = aws.Int64(int64(v))
-			} else if v, ok := bd["iops"].(int); ok && v > 0 && *ebs.VolumeType != "io1" {
+			} else if v, ok := bd["iops"].(int); ok && v > 0 && aws.StringValue(ebs.VolumeType) != "io1" {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:1809]: Can use `ec2.VolumeTypeIo1` here.

File: aws/resource_aws_instance.go [hunk 5]
```diff
@@ -1835,22 +1856,22 @@ func readVolumeTags(conn *ec2.EC2, instanceId string) ([]*ec2.Tag, error) {
 // config determine which one to use in Plan and Apply.
 func readSecurityGroups(d *schema.ResourceData, instance *ec2.Instance, conn *ec2.EC2) error {
 	// An instance with a subnet is in a VPC; an instance without a subnet is in EC2-Classic.
-	hasSubnet := instance.SubnetId != nil && *instance.SubnetId != ""
+	hasSubnet := instance.SubnetId != nil && aws.StringValue(instance.SubnetId) != ""
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:1859]: `aws.StringValue(instance.SubnetId) != ""` will be sufficient as `aws.StringValue` converts `nil` to `""`.

File: aws/resource_aws_instance.go [hunk 6]
```diff
@@ -1835,22 +1856,22 @@ func readVolumeTags(conn *ec2.EC2, instanceId string) ([]*ec2.Tag, error) {
...
 	useID, useName := hasSubnet, !hasSubnet
 
 	// If the instance is in a VPC, find out if that VPC is Default to determine
 	// whether to store names.
-	if instance.VpcId != nil && *instance.VpcId != "" {
+	if instance.VpcId != nil && aws.StringValue(instance.VpcId) != "" {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:1864]: `aws.StringValue(instance.VpcId) != ""` will be sufficient.

File: aws/resource_aws_instance.go [hunk 7]
```diff
@@ -1356,7 +1356,7 @@ func InstanceStateRefreshFunc(conn *ec2.EC2, instanceID string, failStates []str
 	return func() (interface{}, string, error) {
 		instance, err := resourceAwsInstanceFindByID(conn, instanceID)
 		if err != nil {
-			if !isAWSErr(err, "InvalidInstanceID.NotFound", "") {
+			if !isAWSErr(err, ec2.UnsuccessfulInstanceCreditSpecificationErrorCodeInvalidInstanceIdNotFound, "") {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:1359]: I see in `ec2/api.go` `UnsuccessfulInstanceCreditSpecificationErrorCodeInvalidInstanceIdNotFound = "InvalidInstanceID.NotFound"` but even though the constant's value is what we want it's name suggests something to do with an error from getting an instance's credit specification.
I see nothing with leaving the original string here as we do for most other EC2 error codes.

File: aws/resource_aws_instance.go [hunk 8]
```diff
@@ -709,7 +709,7 @@ func resourceAwsInstanceRead(d *schema.ResourceData, meta interface{}) error {
 	if err != nil {
 		// If the instance was not found, return nil so that we can show
 		// that the instance is gone.
-		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "InvalidInstanceID.NotFound" {
+		if isAWSErr(err, ec2.UnsuccessfulInstanceCreditSpecificationErrorCodeInvalidInstanceIdNotFound, "") {
```
Comment (ewbankkit) [inline aws/resource_aws_instance.go:712]: See comment below.

**Comment by ewbankkit** [comment]
Verified (some relevant) acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=TestAccAWSInstance_EbsRootDevice_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 2 -run=TestAccAWSInstance_EbsRootDevice_ -timeout 120m
=== RUN   TestAccAWSInstance_EbsRootDevice_basic
=== PAUSE TestAccAWSInstance_EbsRootDevice_basic
=== RUN   TestAccAWSInstance_EbsRootDevice_ModifySize
=== PAUSE TestAccAWSInstance_EbsRootDevice_ModifySize
=== RUN   TestAccAWSInstance_EbsRootDevice_ModifyType
=== PAUSE TestAccAWSInstance_EbsRootDevice_ModifyType
=== RUN   TestAccAWSInstance_EbsRootDevice_ModifyIOPS
=== PAUSE TestAccAWSInstance_EbsRootDevice_ModifyIOPS
=== RUN   TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination
=== PAUSE TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination
=== RUN   TestAccAWSInstance_EbsRootDevice_ModifyAll
=== PAUSE TestAccAWSInstance_EbsRootDevice_ModifyAll
=== RUN   TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifySize
=== PAUSE TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifySize
=== RUN   TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination
=== PAUSE TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination
=== RUN   TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
=== PAUSE TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
=== CONT  TestAccAWSInstance_EbsRootDevice_basic
=== CONT  TestAccAWSInstance_EbsRootDevice_ModifyAll
--- PASS: TestAccAWSInstance_EbsRootDevice_basic (112.50s)
=== CONT  TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyAll (142.10s)
=== CONT  TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifyDeleteOnTermination (126.42s)
=== CONT  TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifySize
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleDynamicEBSBlockDevices (222.28s)
=== CONT  TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination
--- PASS: TestAccAWSInstance_EbsRootDevice_MultipleBlockDevices_ModifySize (145.75s)
=== CONT  TestAccAWSInstance_EbsRootDevice_ModifyIOPS
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyDeleteOnTermination (116.19s)
=== CONT  TestAccAWSInstance_EbsRootDevice_ModifyType
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyIOPS (156.58s)
=== CONT  TestAccAWSInstance_EbsRootDevice_ModifySize
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifyType (178.96s)
--- PASS: TestAccAWSInstance_EbsRootDevice_ModifySize (146.83s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	717.717s
```

File: aws/resource_aws_instance.go (hunk 2 above)
Comment (anGie44) [reply aws/resource_aws_instance.go:846]: ahh of course 😅, will update

File: aws/resource_aws_instance.go (hunk 7 above)
Comment (anGie44) [reply aws/resource_aws_instance.go:1359]: makes sense 👍 , will revert here and below

**Comment by anGie44** [comment]
thanks @ewbankkit for the review/testing verification! PR now updated with changes from review comments

**Pull request merged**